vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml

```diff
@@ -138,6 +138,94 @@ jobs:
           JAVA_VERSION: $(JAVA_VERSION)
           COMMIT_FROM: $(COMMIT_FROM)
 
+      - task : Docker@2
+        displayName: "Login to ACR"
+        inputs:
+          command: "login"
+          containerRegistry: "Azure IoT ACR"
+
+      - task: PowerShell@2
+        displayName: 'Cert Validation - Setup Certificate Proxy'
+        inputs:
+          targetType: 'inline'
+          script: |
+            Write-Host "============"
+            Write-Host "check docker"
+            Write-Host "============"
+            docker images
+            docker ps -a
+            gci -Path $(Build.SourcesDirectory)/iot-e2e-tests/docker -Recurse -Force -Name
+            Write-Host "==========================="
+            Write-Host "uncompress self-signed cert"
+            Write-Host "==========================="
+            gzip -d -N -S ".bin" $(Build.SourcesDirectory)/iot-e2e-tests/docker/haproxy/haproxy.bin
+            gci -Path $(Build.SourcesDirectory)/iot-e2e-tests/docker -Recurse -Force -Name
+            Write-Host "====================="
+            Write-Host "Docker run containers"
+            Write-Host "====================="
+            docker run -h invalidcertgde1.westus.cloudapp.azure.com --name invalid-gde --expose=443 --expose=5671 --expose=8883 -v $(Build.SourcesDirectory)/iot-e2e-tests/docker/haproxy:/usr/local/etc/haproxy:ro -d aziotacr.azurecr.io/haproxy haproxy -f /usr/local/etc/haproxy/haproxygde.cfg
+            docker run -h invalidcertdps1.westus.cloudapp.azure.com --name invalid-dps --expose=443 --expose=5671 --expose=8883 -v $(Build.SourcesDirectory)/iot-e2e-tests/docker/haproxy:/usr/local/etc/haproxy:ro -d aziotacr.azurecr.io/haproxy haproxy -f /usr/local/etc/haproxy/haproxydps.cfg
+            docker run -h invalidcertiothub1.westus.cloudapp.azure.com --name invalid-hub --expose=443 --expose=5671 --expose=8883 -v $(Build.SourcesDirectory)/iot-e2e-tests/docker/haproxy:/usr/local/etc/haproxy:ro -d aziotacr.azurecr.io/haproxy haproxy -f /usr/local/etc/haproxy/haproxyhub.cfg
+            docker ps -a
+
+      - task: Bash@3
+        displayName: 'Cert Validation - Setup local hostname'
+        name: CVTEST_NET
+        inputs:
+          targetType: 'inline'
+          script: |
+            echo "==============="
+            echo "Inspect network"
+            echo "==============="
+            ip -4 addr
```

Comment:
Script seems to not work on android as is, so this is currently only for the linux builds